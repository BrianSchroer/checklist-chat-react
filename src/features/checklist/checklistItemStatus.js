export const NOT_STARTED                = 'NotStarted';
export const IN_PROGRESS                = 'InProgress';
export const IN_PROGRESS_WITH_ISSUES    = 'InProgressWithIssues';
export const COMPLETED_SUCCESSFULLY     = 'CompletedSuccessfully';
export const COMPLETED_WITH_ISSUES      = 'CompletedWithIssues';
export const COMPLETED_WITH_ERRORS      = 'CompletedWithErrors';
export const CANCELED                   = 'Canceled';
