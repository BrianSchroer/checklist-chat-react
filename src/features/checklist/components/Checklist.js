import React, {PropTypes} from 'react';
import ChecklistItem from './ChecklistItem';

const Checklist = ({checklistItems, OnEditRequest, OnAddRequest}) => {
    return (
        <div className="checklist-table-container">
            <table className="table checklist-table">
                <thead>
                    <tr>
                        <th>&nbsp;</th>
                        <th>&nbsp;</th>
                        <th>Start</th>
                        <th>End</th>
                        <th>Description</th>
                        <th>Performed by</th>
                        <th>&nbsp;</th>
                    </tr>
                </thead>
                {checklistItems.map(item =>
                    <ChecklistItem key={item.sequenceNumber}
                        checklistItem={item}
                        OnEditRequest={function(event) {
                            OnEditRequest(event, item.roomId, item.sequenceNumber);
                        }}/>)}
                <tfoot>
                    <tr>
                        <td colSpan="7">
                            <button className="btn btn-primary pull-right checklist-item-button"
                                onClick={OnAddRequest}>
                                Add a new checklist item
                            </button>
                        </td>
                    </tr>
                </tfoot>
            </table>
        </div>
    );
};

Checklist.propTypes = {
    checklistItems: PropTypes.arrayOf(PropTypes.object).isRequired,
    OnEditRequest: PropTypes.func.isRequired,
    OnAddRequest: PropTypes.func.isrRequired
};

export default Checklist;
