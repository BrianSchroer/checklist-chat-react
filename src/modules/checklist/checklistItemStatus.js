export const NOT_STARTED                = 'NotStarted';
export const IN_PROGRESS                = 'InProgress';
export const IN_PROGRESS_WITH_ISSUES    = 'InProgressWithIssues';
export const COMPLETED_SUCCESSFULLY     = 'CompletedSuccessfully';
export const COMPLETED_WITH_ISSUES      = 'CompletedWithIssues';
export const COMPLETED_WITH_ERRORS      = 'CompletedWithErrors';
export const CANCELED                   = 'Canceled';

export const options = [
    { value: NOT_STARTED, text: 'Not Started' },
    { value: IN_PROGRESS, text: 'In Progress' },
    { value: IN_PROGRESS_WITH_ISSUES, text: 'In Progress With Issues' },
    { value: COMPLETED_SUCCESSFULLY, text: 'Completed Successfully' },
    { value: COMPLETED_WITH_ISSUES, text: 'Completed with Issues' },
    { value: COMPLETED_WITH_ERRORS, text: 'Completed with Errors' },
    { value: CANCELED, text: 'Canceled' }
];
