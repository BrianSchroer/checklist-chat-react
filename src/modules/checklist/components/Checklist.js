import React from 'react';
import PropTypes from 'prop-types';
import ChecklistAddInvitation from './ChecklistAddInvitation';
import ChecklistItem from './ChecklistItem';

const Checklist = ({ checklistItems, onEditRequest }) => {
  const hasItems = checklistItems.length > 0;
  const hasNoItems = !hasItems;

  return (
    <div className="checklist-table-container">
      <table className="table checklist-table">
        <thead>
          <tr>
            <th />
            <th />
            <th>Start</th>
            <th>End</th>
            <th>Description</th>
            <th>Performed by</th>
          </tr>
        </thead>
        {hasItems &&
          checklistItems.map(item => (
            <ChecklistItem
              key={item.id}
              checklistItem={item}
              onEditRequest={function(event) {
                onEditRequest(event, item.roomId, item.sequenceNumber);
              }}
            />
          ))}
        {hasNoItems && <ChecklistAddInvitation />}
      </table>
    </div>
  );
};

Checklist.propTypes = {
  checklistItems: PropTypes.arrayOf(PropTypes.object).isRequired,
  onEditRequest: PropTypes.func.isRequired
};

export default Checklist;
