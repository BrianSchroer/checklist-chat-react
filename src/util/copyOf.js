export default original => Object.assign({}, original);
