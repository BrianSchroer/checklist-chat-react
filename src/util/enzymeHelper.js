const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
