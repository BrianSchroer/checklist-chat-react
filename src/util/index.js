import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
