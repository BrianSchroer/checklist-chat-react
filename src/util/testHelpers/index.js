import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
