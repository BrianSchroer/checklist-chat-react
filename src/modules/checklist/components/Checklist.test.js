import React from 'react';
import {snapshotHelper} from '../../../util/testHelpers';
import Checklist from './Checklist';
import * as checklistItemStatus from '../checklistItemStatus';

const defaultProps = {
    checklistItems: [],
    onEditRequest: () => {}
};

function overrideProps(propOverrides) {
    return Object.assign({}, defaultProps, propOverrides);
}

function assertSnapshotMatch(propOverrides = {}) {
    snapshotHelper.assertMatch(<Checklist {...overrideProps(propOverrides)} />);
}

describe('Checklist', () => {
    describe('with no existing items', () => {
        it('should render correctly', () => {
            assertSnapshotMatch({checklistItems: []});
        });
   });

    describe('with existing items', () => {
        it('should render correctly', () => {
            assertSnapshotMatch({
                checklistItems: [
                    { id: 1, sequenceNumber: 1, status: checklistItemStatus.IN_PROGRESS },
                    { id: 2, sequenceNumber: 2, status: checklistItemStatus.NOT_STARTED }
                ]
            });
        });
   });
});
