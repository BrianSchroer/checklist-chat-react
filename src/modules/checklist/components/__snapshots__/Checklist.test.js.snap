// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checklist with existing items should render correctly 1`] = `
<div
  className="checklist-table-container"
>
  <table
    className="table checklist-table"
  >
    <thead>
      <tr>
        <th>
           
        </th>
        <th>
           
        </th>
        <th>
          Start
        </th>
        <th>
          End
        </th>
        <th>
          Description
        </th>
        <th>
          Performed by
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="checklist-item-row"
      >
        <td>
          <button
            className="btn btn-primary btn-sm checklist-item-button"
            onClick={[Function]}
            title="Edit this item"
          >
            1
          </button>
        </td>
        <td>
          <i
            className="glyphicon glyphicon-repeat success-icon spinning-icon"
            title="in progress"
          />
        </td>
        <td
          className="timestamp"
        >
          <div
            className="planned-vs-actual"
          >
            <div
              className="planned-value-missing"
            >
              
            </div>
            <div
              className="actual-value-missing"
            >
              
            </div>
          </div>
        </td>
        <td
          className="timestamp"
        >
          <div
            className="planned-vs-actual"
          >
            <div
              className="planned-value-missing"
            >
              
            </div>
            <div
              className="actual-value-missing"
            >
              
            </div>
          </div>
        </td>
        <td />
        <td />
      </tr>
    </tbody>
    <tbody>
      <tr
        className="checklist-item-row"
      >
        <td>
          <button
            className="btn btn-primary btn-sm checklist-item-button"
            onClick={[Function]}
            title="Edit this item"
          >
            2
          </button>
        </td>
        <td>
          <i
            className="glyphicon glyphicon-unchecked"
            title="not started"
          />
        </td>
        <td
          className="timestamp"
        >
          <div
            className="planned-vs-actual"
          >
            <div
              className="planned-value-missing"
            >
              
            </div>
            <div
              className="actual-value-missing"
            >
              
            </div>
          </div>
        </td>
        <td
          className="timestamp"
        >
          <div
            className="planned-vs-actual"
          >
            <div
              className="planned-value-missing"
            >
              
            </div>
            <div
              className="actual-value-missing"
            >
              
            </div>
          </div>
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Checklist with no existing items should render correctly 1`] = `
<div
  className="checklist-table-container"
>
  <table
    className="table checklist-table"
  >
    <thead>
      <tr>
        <th>
           
        </th>
        <th>
           
        </th>
        <th>
          Start
        </th>
        <th>
          End
        </th>
        <th>
          Description
        </th>
        <th>
          Performed by
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
        </td>
        <td>
           
        </td>
        <td
          className="text-success"
          colSpan="4"
        >
          <h4>
            This chat has no checklist items.
          </h4>
          <h4>
            Use the "Add Checklist Item" button below to add one.
          </h4>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
